File: app/src/main/java/com/qimian233/ztool/hook/base/AppHookModule.java
package com.qimian233.ztool.hook.base;

import io.github.libxposed.api.XposedModuleInterface;

/**
 * 普通 App Hook 模块基类。
 * <p>
 * 继承此类后 IDE 会提示实现 {@link #handleLoadPackage}，无需关心
 * {@link #handleSystemServerStarting}。
 * </p>
 */
public abstract class AppHookModule extends BaseHookModule {

    /**
     * 执行 App 包加载时的 Hook 操作（<b>abstract</b>，便于 IDE 自动补全）。
     */
    @Override
    public abstract void handleLoadPackage(XposedModuleInterface.PackageLoadedParam param) throws Throwable;
}

File: app/src/main/java/com/qimian233/ztool/hook/base/AppHookModule.kt
package com.qimian233.ztool.hook.base

import io.github.libxposed.api.XposedModuleInterface

/**
 * Base class for normal app Hook modules (Kotlin).
 * <p>
 * After extending this class the IDE will prompt to implement
 * [handleLoadPackage]; [handleSystemServerStarting] is irrelevant here.
 * </p>
 */
abstract class AppHookModule : BaseHookModule() {

    /**
     * Performs Hook operations on app package load (**abstract**, for IDE autocompletion).
     * <p>Carries [@Throws](Throwable::class) so Java subclasses can keep declaring {@code throws Throwable}.</p>
     */
    @Throws(Throwable::class)
    abstract override fun handleLoadPackage(param: XposedModuleInterface.PackageLoadedParam)
}
